### PR TITLE
(DOCSP-20060): Kotlin SDK - Sync

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
### Jira

- https://jira.mongodb.org/browse/DOCSP-20060
- Renamed "Kotlin Multiplatform SDK" to "Kotlin SDK"
- Added Sync Examples
- Scrapped actual tests for Sync Examples. If anybody knows how to get these tests running in JVM KMM tests... I would very much appreciate a pointer. I got them working briefly, but an update to coroutines, the KMM plugin, kotlin, or perhaps the SDK snapshots broke it soon after (hard to tell).

### Staged Changes (Requires MongoDB Corp SSO)

- [Kotlin SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/kotlin-sdk-sync/sdk/kotlin)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/examples/generated/kotlin/Greeting.codeblock.configure-a-synced-realm.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+val app = App.create(YOUR_APP_ID)
+runBlocking() {
+    val user = app.login(Credentials.anonymous())
+    val config = SyncConfiguration.Builder(user, PARTITION)
+        // specify name so realm doesn't just use the "default.realm" file for this user
```
Comment (cmelchior) [inline source/examples/generated/kotlin/Greeting.codeblock.configure-a-synced-realm.kt:5]: For partition-based realms, we shouldn't be using `default.realm` but something like `s_<partitionValue>.realm` instead. So this shouldn't be needed.

File: source/examples/generated/kotlin/Greeting.codeblock.landing-page-update.kt [hunk 2]
```diff
@@ -3,8 +3,8 @@ val realm = Realm.open(config)
 // start a write transaction
 realm.writeBlocking {
     // get a frog from the database to update
-    val frog = objects<Frog>()
-        .query("name == $0 LIMIT(1)", "Benjamin Franklin")
+    val frog = query<Frog>()
+        .query("name == $0 LIMIT(1)", "Benjamin Franklin").find()
```
Comment (cmelchior) [inline source/examples/generated/kotlin/Greeting.codeblock.landing-page-update.kt:7]: I would rewrite this to:

```
val frog = query<Frog>("name == $0 LIMIT(1)", "Benjamin Franklin").find()
```

File: source/sdk/kotlin.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+   
+.. include:: /includes/note-alpha-sdk.rst
+
+The {+service+} :github:`Kotlin SDK <realm/realm-kotlin>`
```
Comment (cmelchior) [inline source/sdk/kotlin.txt:26]: Maybe rephrase this so it is very clear that we also support standalone Android apps?

File: source/sdk/kotlin.txt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Get Started
+-----------
+
+To start using the {+service+} Kotlin SDK
+in your KMM application, see :ref:`the installation guide
+<kotlin-install>` to add the Kotlin SDK
+dependency. Once you've added the SDK to your application, check out the
+:ref:`Quick Start <kotlin-client-quick-start-local>`.
+
+- :doc:`Install Realm for Kotlin </sdk/kotlin/install>`
```
Comment (cmelchior) [inline source/sdk/kotlin.txt:78]: Probably fine for now, but I would split this into `Install Realm for Multiplatform` and `Install Realm for Android` later.

**Review by cmelchior — COMMENTED** [review]
Nice..only had a few minor comments.

File: source/examples/generated/kotlin/Greeting.codeblock.configure-a-synced-realm.kt (hunk 1 above)
Comment (nathan-contino-mongo) [reply source/examples/generated/kotlin/Greeting.codeblock.configure-a-synced-realm.kt:5]: Hm, I think this is an SDK bug, then -- I definitely had to specify a name, or I'd end up with an error because completely different realms were trying to use the "default.realm" file. This was my workaround for that.
